### ISSUE TITLE
Cannot build demo maven build error

### ISSUE THREAD

**Issue by WerVbn** [eclipse/kapua#598]
when I running ./start-demo.sh, maven throws the following build error:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-clean-plugin:2.5:clean (default-clean) on project kapua-build-tools: Failed to clean project: Failed to delete /kapua/build-tools/target/classes/checkstyle/kapua.suppressions.xml -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
VM must be created before running this command. Run `vagrant up` first.

$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'base' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: >= 0
==> default: Box file was not detected as metadata. Adding it directly...
==> default: Adding box 'base' (v0) for provider: virtualbox
    default: Downloading: base
An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

Couldn't open file kapua/dev-tools/src/main/bin/base

```
How can I fix that?

**Comment by Coduz** [comment]
Hi @tesfel 

Can I suggest to run `git clean -dxf` from the Kapua project root directory and retry to start the demo box from `dev-tools/src/main/bin/` ?

Thanks, 

_Alberto_

**Comment by tesfel** [comment]
Hello,

thanks for your reply. 
`./start-demo.sh` fails now with the following error:

```
[INFO] Total time: 03:15 min
[INFO] Finished at: 2017-05-29T15:58:14+02:00
[INFO] Final Memory: 158M/506M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.1:exec (vagrant_console_deploy) on project kapua-assembly-api: Result of /bin/sh -c cd /kapua/assembly/api/../../dev-tools/src/main/vagrant/demo && ./deployScript/deploy-apps.sh execution is: '1'. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :kapua-assembly-api
VM must be created before running this command. Run `vagrant up` first.
```

**Comment by Coduz** [comment]
Hi @tesfel ,

it is likely that you need to pull the latest develop. 
Can you please do that? You don't need to run again `git clean -dxf` but do it to have a clean repo.

Thanks, 

_Alberto_

**Comment by tesfel** [comment]
I think I pulled the latest version with

`$ git clone https://github.com/eclipse/kapua`

**Comment by Coduz** [comment]
From the network graph it seems that you are at https://github.com/eclipse/kapua/commit/343d067fe911741627d56bf802ed1194225889f6 on 25 May.

**Comment by tesfel** [comment]
Still does not work. Do you got any other idea how to solve this issue?

**Comment by ctron** [comment]
I would suggest to go with the docker version instead of vagrant: https://github.com/eclipse/kapua/tree/develop/assembly#run

**Comment by Coduz** [comment]
Hi @tesfel ,

Is this still a problem??
Otherwise we will close this issue.

Regards, 

_Alberto_

**Comment by kamir** [comment]
I was running into the same issue.

Cloned latest kapua repo ...
Checkout branch release-0.3.x 

mvn clean compile  -  works fine

The start-demo.sh script fails.
Especially:

mvn clean install -PdeployVagrant,console -D skipTests

Gives me:

[INFO] kapua-rest-api-web ................................. FAILURE [  0.250 s]
[INFO] kapua-job-engine-jbatch ............................ SKIPPED
[INFO] kapua-client ....................................... SKIPPED
[INFO] kapua-client-gateway ............................... SKIPPED
[INFO] Eclipse Kapua :: Gateway Client :: API ............. SKIPPED
[INFO] Eclipse Kapua :: Gateway Client :: SPI ............. SKIPPED
[INFO] Eclipse Kapua :: Gateway Client :: Provider ........ SKIPPED
[INFO] Eclipse Kapua :: Gateway Client :: Provider :: Core MQTT SKIPPED
[INFO] Eclipse Kapua :: Gateway Client :: Provider :: FUSE MQTT SKIPPED
[INFO] Eclipse Kapua :: Gateway Client :: Provider :: Eclipse Paho SKIPPED
[INFO] Eclipse Kapua :: Gateway Client :: Profile :: Eclipse Kura SKIPPED
[INFO] Eclipse Kapua :: Gateway Client :: Features ........ SKIPPED
[INFO] Eclipse Kapua :: Gateway Client :: Features :: Karaf SKIPPED
[INFO] kapua-assembly ..................................... SKIPPED
[INFO] kapua-assembly-java-base ........................... SKIPPED
[INFO] kapua-assembly-jetty-base .......................... SKIPPED
[INFO] kapua-assembly-sql ................................. SKIPPED
[INFO] kapua-assembly-broker .............................. SKIPPED
[INFO] kapua-assembly-api ................................. SKIPPED
[INFO] kapua-console ...................................... SKIPPED
[INFO] kapua-console-module ............................... SKIPPED
[INFO] kapua-console-module-api ........................... SKIPPED
[INFO] kapua-console-module-about ......................... SKIPPED
[INFO] kapua-console-module-authentication ................ SKIPPED
[INFO] kapua-console-module-authorization ................. SKIPPED
[INFO] kapua-console-module-user .......................... SKIPPED
[INFO] kapua-console-module-data .......................... SKIPPED
[INFO] kapua-console-module-device ........................ SKIPPED
[INFO] kapua-console-module-account ....................... SKIPPED
[INFO] kapua-console-module-job ........................... SKIPPED
[INFO] kapua-console-module-tag ........................... SKIPPED
[INFO] kapua-console-module-welcome ....................... SKIPPED
[INFO] kapua-console-core ................................. SKIPPED
[INFO] kapua-console-web .................................. SKIPPED
[INFO] kapua-assembly-console ............................. SKIPPED
[INFO] kapua-dev-tools .................................... SKIPPED
[INFO] kapua-cucumber-reports ............................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 38.141 s
[INFO] Finished at: 2017-09-22T17:03:25+02:00
[INFO] Final Memory: 210M/2180M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.0.1:unpack (unpack-swagger-ui) on project kapua-rest-api-web: Unable to find/resolve artifact. Failure to find org.eclipse.kapua.external:swagger-ui:jar:2.1.4-CQ-10792 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :kapua-rest-api-web

**Comment by kamir** [comment]
One reason for failures in the start-demo.sh script was a missing mount point in the base image.
I solved this with adding the guest-additions using a vagrant plugin:

**vagrant plugin install vagrant-vbguest**

The missing artifact is still a problem:
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.0.1:unpack (unpack-swagger-ui) on project kapua-rest-api-web: Unable to find/resolve artifact. Failure to find org.eclipse.kapua.external:swagger-ui:jar:2.1.4-CQ-10792 in https://repo.maven.apache.org/maven2 was cached in the lo

**Comment by kamir** [comment]
OK, where to get the swagger-ui JAR?

There is a folder "external/swagger-ui-lib" in Kapua. But the pom.xm is not referenced in the main pom.xml.

OK, lets test again .... Yeah. 
_Finally it worked._

**Summary:**

1) Manage VitualBox guest additions:
**_vagrant plugin install vagrant-vbguest_** 

2) Add this to your pom file:

    <modules>
...
        
        <module>external</module>
...
    </modules>

**Comment by Coduz** [comment]
Hi @kamir ,

Yes after the 0.2.0 release we had to change a bit the project modules layout. 

For release purposes we had to remove the `external` modules from the root `pom.xml`. 
Now is required to first build the `external` dependency and then the rest of the project.

We will update the documentation to reflect these changes where we missed them.

As per solution commands to build correctly kapua now are

```
mvn clean install -f external/pom.xml
mvn clean install 
```

or with the Web Admin Console

```
mvn clean install -f external/pom.xml
mvn clean install -Pconsole
```
I'll comment on your PR pointing to this answer.

Regards,

_Alberto_
